File: sentimental-credit/credit.py
from cs50 import get_int


def main():
    while True:
        num = get_int("What's your credit card number? ")
        if num >= 0:
            break  # look for the valid case here and break

    if check_validity(num):
        print_brand(num)
    else:
        print("INVALID")


def check_validity(ccn):
    return checksum(ccn)


def checksum(ccn):

    sum = 0
    for i in range(len(str())):
        if i % 2 == 0:
            sum += ccn % 10
        else:
            digit = 2 * (ccn % 10)
            sum += digit // 10 + digit % 10
        ccn //= 10
    return sum % 10 == 0


def print_brand(ccn):

    if (ccn >= 34e13 and ccn < 35e13) or (ccn >= 37e13 and ccn < 38e13):
        print("AMEX")
    elif (ccn >= 4e13 and ccn < 5e12) or (ccn >= 4e15 and ccn < 5e15):
        print("VISA")
    elif ccn >= 51e14 and ccn < 56e14:
        print("MASTERCARD")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()

# from cs50 import get_string

# num = get_string("What's your credit card number? ")
# reverse = num[::-1]
# total_sum = sum ( [ (int(x) * 2) // 10 + ( (int(x) * 2) % 10) for x in reverse[1::2]]) + sum ( [int(x) for x in reverse[0::2]] )

# if total_sum % 10 == 0:
#     if len(num) == 15 and num[0:2] in ['34', '37']:
#         print("AMEX")
#     elif len(num) == 16 and num[0:2] in ['51', '52', '53', '54', '55']:
#         print("MASTERCARD")
#     elif len(num) == 13 or 16 and num[0] in ['4']:
#     # elif len(num) in [13, 16] and num[0] == '4': # This also works
#         print("VISA")
#     else:
#         print("INVALID")
# else:
#     print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float sixteenyes;
float visasixteenyes;
float amexyes;
float visayes;
long secondtolastnum;
long num;
int digit;
int digit1;
int digit2;
int sum;
int sum1;
int sum2;
int count;
int func(long n);
int main(void)
{
//PROMPT FOR INPUT
    do
    {
        num = get_long("What's your credit card number? ");
    }
    while (num < 1);

    visayes = num / 1000000000000;
    visasixteenyes = num / 1000000000000000;
    amexyes = num / 10000000000000;
    sixteenyes = num / 100000000000000;

// COUNT NUMBER OF DIGITS USING COUNT FUNCTION
    count = 0;
    count = func(num);

    if (count < 13)
    {
        printf("INVALID\n");
        return 0;
    }
    else if (count == 14)
    {
        printf("INVALID\n");
        return 0;
    }
    else if (count > 16)
    {
        printf("INVALID\n");
        return 0;
    }

//CHECK NUMBER LENGTH OF 15
    if (count == 15)
    {
        if (34 == amexyes)
        {
            printf("AMEX\n");
            return 0;
        }
        if (37 == amexyes)
        {
            printf("AMEX\n");
            return 0;
        }
        else
        {
            printf("INVALID\n");
            return 0;
        }
    }

//ADDING THE SUMS OF LUHN'S ALGO
    secondtolastnum = num / 10;
    while (secondtolastnum > 0)
    {
        digit = (secondtolastnum % 10) * 2;
        secondtolastnum = secondtolastnum / 100;

        if (digit > 9)
        {
            digit1 = digit % 10;
            digit2 = digit / 10 % 10;
            digit = digit1 + digit2;
        }
        sum1 = sum1 + digit;
    }

    while (num > 0)
    {
        digit = (num % 10);
        num = num / 100;

        if (digit > 9)
        {
            digit1 = digit % 10;
            digit2 = digit / 10 % 10;
            digit = digit1 + digit2;
        }
        sum2 = sum2 + digit;
    }
    sum = sum1 + sum2;

// CHECK FOR NUMBER LENGTH OF 13 (VISA)
    if (count == 13)
    {
        while (4 == visayes)
        {
            if (sum % 10 == 0)
            {
                printf("VISA\n");
                return 0;
            }
            else
            {
                printf("INVALID\n");
                return 0;
            }
        }
    }

// CHECK FOR NUMBER LENGTH OF 16 (VISA)
    if (count == 16)
    {
        if (4 == visasixteenyes)
        {
            if (sum % 10 == 0)
            {
                printf("VISA\n");
                return 0;
            }
        }
    }
// CONDITIONS FOR MASTERCARD TO BE TRUE
    if (51 == sixteenyes || 52 == sixteenyes || 53 == sixteenyes || 54 == sixteenyes || 55 == sixteenyes)
    {
        if (sum % 10 == 0)
        {
            printf("MASTERCARD\n");
            return 0;
        }
    }
    else
    {
        printf("INVALID\n");
        return 0;
    }
}

int func(long n)
{
    int counter = 0; // variable declaration
    while (n != 0)
    {
        n = n / 10;
        counter++;
    }
    return counter;
}